### PR TITLE
Add support for multi-module projects

### PR DESCRIPTION
**Pull request by shadaj**
TODO:
- [x] Handle imports by creating aliases
- [x] Handle aliases for custom types
- [x] ~Support wildcard imports~ (#31)

### CODE REVIEW

File: core/src/ast/modules.rs [hunk 1]
```diff
@@ -130,12 +137,36 @@ impl Module {
...
         Module {
             name: input.ident.to_string(),
+            imports,
             declared_types: custom_types_by_name,
             sub_modules,
         }
     }
 }
 
+fn extract_imports(base_path: &Path, use_tree: &UseTree, out: &mut Vec<(Path, String)>) {
```
Comment (Manishearth) [inline core/src/ast/modules.rs:147]: I would only extract tagged imports, other imports should just be ignored IMO.

File: core/src/ast/modules.rs [hunk 2]
```diff
@@ -2,13 +2,14 @@ use serde::{Deserialize, Serialize};
...
 use quote::ToTokens;
-use syn::{ImplItem, Item, ItemMod};
+use syn::{ImplItem, Item, ItemMod, UseTree};
 
 use super::{CustomType, Method, ModSymbol, OpaqueStruct, Path, Struct, TypeName};
 
 #[derive(Clone, Serialize, Deserialize, Debug)]
 pub struct Module {
     pub name: String,
+    pub imports: Vec<(Path, String)>,
```
Comment (Manishearth) [inline core/src/ast/modules.rs:12]: If we only extract tagged imports, should these just add to the list of declared types?

Comment (shadaj) [reply core/src/ast/modules.rs:12]: We want to support imports of other modules, so we can't assume that the import always is to a type.

File: core/src/ast/modules.rs (hunk 1 above)
Comment (shadaj) [reply core/src/ast/modules.rs:147]: Hmm, is there any advantage from the Diplomat side for doing this? We handle unsupported imports by just not adding symbols at resolution time anyways, so was thinking of only requiring annotations for DSTs to reduce noise.

**Review by Manishearth — APPROVED** [review]

**Pull request merged**